src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -832,3 +1000,56 @@ def __removed_older_complete_status_files(self, status_folder):
+    def __calc_package_payload_size_on_disk(self, package_list):
+        """ Calculate final package list size in bytes (because of escape chars) """
+        first_json_dump = json.dumps(package_list)
+        return len(json.dumps(first_json_dump).encode("utf-8"))
+
+    def __size_of_constant_status_data(self, complete_status_file_payload_json, assessment_status_index, installation_status_index):
+        """ Get the size in bytes of the complete_status_file without packages data  """
+        status_file_no_list_data = complete_status_file_payload_json
+        if assessment_status_index is not None:
+            assessment_msg_without_packages = self.__update_substatus_msg(substatus_msg=self.__assessment_substatus_msg_copy, substatus_msg_patches=[])
+            status_file_no_list_data['status']['substatus'][assessment_status_index]['formattedMessage']['message'] = json.dumps(assessment_msg_without_packages)
+
+        if installation_status_index is not None:
+            installation_msg_without_packages = self.__update_substatus_msg(substatus_msg=self.__installation_substatus_msg_copy, substatus_msg_patches=[])
+            status_file_no_list_data['status']['substatus'][installation_status_index]['formattedMessage']['message'] = json.dumps(installation_msg_without_packages)
+        return self.__calc_status_size_on_disk(json.dumps(status_file_no_list_data))
+
+    def __get_substatus_index(self, substatus_list_name, substatus_list):
+        """ Gets the index of an operation substatus in the overall extension status for further operations """
+        for substatus_index, substatus_name in enumerate(substatus_list):
+            if substatus_name['name'] == substatus_list_name:
+                return substatus_index
+        return None
+
+    def __recompose_truncated_status_file(self, truncated_status_file, truncated_package_list, substatus_message, substatus_index):
+        """ Recompose final truncated status file version """
+        code, _ = self.__get_errors_from_substatus(substatus_msg=substatus_message)
+
+        # Check for existing errors before recompose
+        if code != Constants.PatchOperationTopLevelErrorCode.ERROR:
+            self.composite_logger.log_debug("Packages in substatus have been truncated hence updating status to [status={0}]".format(Constants.STATUS_WARNING))
+            truncated_status_file['status']['substatus'][substatus_index]['status'] = Constants.STATUS_WARNING.lower()      # Update substatus status to warning
+
+        self.composite_logger.log_verbose("Recompose truncated substatus")
+        truncated_substatus_message = self.__update_substatus_msg(substatus_msg=substatus_message, substatus_msg_patches=truncated_package_list)
+
+        truncated_status_file['status']['substatus'][substatus_index]['formattedMessage']['message'] = json.dumps(truncated_substatus_message)
+        return truncated_status_file
+
+    def __update_substatus_msg(self, substatus_msg, substatus_msg_patches):
```

Comment:
__update_patches_in_substatus()

Since this function is only updating patches within the substatus.

Reply:
can't update to this because the purpose of the function is updating the entire message json. if i change it then in the later prs there will be ask to change it back, pending

Reply:
pending

Reply:
Like it was discussed before, both offline and as PR feedback comments, never keep anything pending for later PRs. All code changes going in, in one PR should be final and related to only it's current context. Things can be changed in future, if needed. Please make this change

Reply:
Maintaining exclusivity between PRs: Please change the function name, since it is only updating patches at this point. If the function changes in future, and those changes demand a name change, that will be applicable and implemented in those PRs

Reply:
done